src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -2769,22 +3017,26 @@ public override IReadOnlyList<SyntaxNode> GetStatements(SyntaxNode declaration)
 
         public override SyntaxNode WithStatements(SyntaxNode declaration, IEnumerable<SyntaxNode> statements)
         {
+            return Isolate(declaration, d => WithStatementsInternal(d, statements));
+        }
+
+        private SyntaxNode WithStatementsInternal(SyntaxNode declaration, IEnumerable<SyntaxNode> statements)
+        {
             BlockSyntax body = CreateBlock(statements);
-            BlockSyntax somebody = statements != null ? body : null;
-            SyntaxToken semicolon = statements == null ? SyntaxFactory.Token(SyntaxKind.SemicolonToken) : default(SyntaxToken);
+            SyntaxToken semicolon = default(SyntaxToken);
```

Comment:
Gotcha.  I would personally prefer that people could do it just by passing null.  But we'll have to pick a default one way or the other, and i don't mind the default you've chosen here.